eox_tenant/admin.py

```diff
@@ -67,6 +67,79 @@ def ednx_signal(self, microsite):
             return str(error)
 
 
+class TenantConfigAdmin(admin.ModelAdmin):
+    """
+    Tenant config model admin
+    """
+    list_display = [
+        'external_key',
+        'domains',
+        'sitename',
+        'template_dir',
+        'course_org_filter',
+        'ednx_signal',
+    ]
+    readonly_fields = (
+        'sitename',
+        'template_dir',
+        'course_org_filter',
+        'ednx_signal',
+    )
+    search_fields = ('key', 'route__domain', 'lms_configs', )
+
+    def sitename(self, tenant_config):
+        """
+        Read only method to calculate sitename attribute from config model.
+        """
+        # pylint: disable=broad-except
+        try:
+            return tenant_config.lms_configs.get('SITE_NAME', "NOT CONFIGURED")
+        except Exception as error:
+            return str(error)
```

Comment:
If you are returning str(error), this means that this error is going to be displayed in the django admin console?

Reply:
it will be displayed on the django admin model interface. Specifically, when listing TenantConfig instances.

Reply:
and is this the behavior we want?

Reply:
Well, this is something that should not happen. But if so, we don't want the whole admin list failing

Reply:
Here are we expecting to have a different exception than AttributeError? considering the case when lms_configs is not defined. On the other hand, can we have a default value in lms_configs like an empty dict?